src/Storages/StorageDictionary.cpp

```diff
@@ -128,37 +167,135 @@ Pipe StorageDictionary::read(
     const Names & column_names,
     const StorageMetadataPtr & /*metadata_snapshot*/,
     SelectQueryInfo & /*query_info*/,
-    ContextPtr context,
+    ContextPtr local_context,
     QueryProcessingStage::Enum /*processed_stage*/,
     const size_t max_block_size,
     const unsigned /*threads*/)
 {
-    auto dictionary = context->getExternalDictionariesLoader().getDictionary(dictionary_name, context);
+    auto dictionary = getContext()->getExternalDictionariesLoader().getDictionary(dictionary_name, local_context);
     auto stream = dictionary->getBlockInputStream(column_names, max_block_size);
     /// TODO: update dictionary interface for processors.
     return Pipe(std::make_shared<SourceFromInputStream>(stream));
 }
 
+void StorageDictionary::drop()
+{
+    removeDictionaryConfigurationFromRepository();
```

Comment:
Seems like it's redundant, because `shutdown()` must be called before `drop()`

Reply:
Fixed.